### PR TITLE
chore (starter): expose configurable proxy port

### PR DESCRIPTION
**Pull request by kevin-dp**
#529 introduces a proxy that runs on a new port (65432 in Electric's Docker container).
This PR does the following:
- exposes the port in the Docker compose files and in the `docker run` commands
- modifies the `db:psql` and `db:migrate` commands to go through the proxy (i.e. via that port)
- modifies the starter to make that port configurable

For now, in order to test this PR, you need to checkout this PR and then:
- build Electric (`cd components/electric && make docker-build`) 
- build the starter (`cd examples/starter && pnpm install && pnpm build && npm link`)
- create an app `create-electric-app my-app`
- modify your app to use the local Electric image (because it has not yet been published to Docker hub)
   i.e. in `my-app/backend/compose/.envrc` change the `ELECTRIC_IMAGE` env var to be `electric:local-build`.

### CODE REVIEW

File: examples/expo/db/util.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ shell.config.silent = true // don't log output of child processes
 // So we fetch this host port and use it in the default url.
 const pgPort = fetchHostPortPG() ?? 5432
 const appName = fetchAppName() ?? 'electric'
-const DEFAULT_URL = `postgresql://postgres:password@localhost:${pgPort}/${appName}`
+const DEFAULT_URL = `postgresql://postgres:password@${appName}.localhost:${pgPort}/electric`
```
Comment (alco) [inline examples/expo/db/util.js:13]: How does this work when `localhost` does not support subdomains?

File: examples/todoMVC/.idea/workspace.xml
Comment (alco) [inline examples/todoMVC/.idea/workspace.xml:1]: Added by mistake?

**Comment by alco** [comment]
@kevin-dp I have rewritten this PR to only update the starter. The changes you've made to other example apps can be found in [alco/update-examples-wip](https://github.com/electric-sql/electric/compare/kevindp/vax-1180-expose-proxy-port...alco/update-examples-wip).

I have tested these changes by following the script in https://electric-sql.slab.com/posts/testing-the-starter-template-3gtk1nlh. Everything is working great except for one thing: `yarn db:migrate` fails the first one or more times I run it. @magnetised this seems to be related to the introduction of the proxy, see below.

When I run `yarn db:migrate` in an app created with the new starter, it just hangs for 60 seconds, then exists:

```
$ yarn db:migrate
yarn run v1.22.19
warning package.json: No license field
warning ../../../package.json: No license field
$ node ./db/migrate.js
Connecting to proxy at postgresql://electric@localhost:65433/app_2
warning package.json: No license field                                                                                                                                                                              
warning ../../../package.json: No license field                                                                                                                                                                     

>>>>>>>>>> it hangs here for 60 seconds <<<<<<<<<<<<<<

Error: Connection terminated unexpectedly
    at /home/alco/code/tmp/app_2/node_modules/pg/lib/client.js:526:17
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async executeQueryInternal (/home/alco/code/tmp/app_2/node_modules/@databases/pg/lib/Driver.js:360:25)
    at async PgDriver._executeQuery (/home/alco/code/tmp/app_2/node_modules/@databases/pg/lib/Driver.js:199:29)
    at async PgDriver.executeAndReturnLast (/home/alco/code/tmp/app_2/node_modules/@databases/pg/lib/Driver.js:226:16)
    at async Transaction.query (/home/alco/code/tmp/app_2/node_modules/@databases/shared/lib/BaseTransaction.js:59:24)
    at async Object.updateVersion (/home/alco/code/tmp/app_2/node_modules/@databases/pg-migrations/lib/PostgresDatabaseEngine.js:81:44)
    at async /home/alco/code/tmp/app_2/node_modules/@databases/migrations-base/lib/runCommand.js:30:13
    at async /home/alco/code/tmp/app_2/node_modules/@databases/pg-migrations/lib/PostgresDatabaseEngine.js:54:20
    at async txInternal (/home/alco/code/tmp/app_2/node_modules/@databases/shared/lib/utils.js:65:22)
 Failed to connect to the DB. Exit code: 1 
Done in 61.45s.
```

Nothing gets logged in Postgres.

Electric log:
```
app_2-electric-1  | 14:32:00.123 pid=<0.3408.0> proxy_session_id=3 [warning] Not validating user "electric"
app_2-electric-1  | 14:32:00.126 pid=<0.3408.0> proxy_session_id=3 [info] Initialising injector in capture mode %Electric.Postgres.Proxy.Injector.Electric{}
app_2-electric-1  | 14:32:00.150 pid=<0.3408.0> proxy_session_id=3 [info] Upstream connection is ready to accept queries

>>>>>>>>>> no output for 60 seconds <<<<<<<<<<<<<<

app_2-postgres-1  | 2023-10-18 14:33:00.214 GMT [100] LOG:  unexpected EOF on client connection with an open transaction
```

It usually succeeds to run migrations on the 2nd or 3rd try. Looks like some sort of race getting in the way of connection setup between the client and the proxy.

**Comment by alco** [comment]
@magnetised

<details>
<summary>Electric's log with LOG_LEVEL=debug and PROXY_TRACING_ENABLE=true</summary>

```
app_2-electric-1  | [1] 🠜 client [%PgProtocol.Message.SSLRequest{}]
app_2-electric-1  | [1] 🠞 client "N"
app_2-electric-1  | [3] 🠜 client [
app_2-electric-1  |   %PgProtocol.Message.StartupMessage{
app_2-electric-1  |     version: {3, 0},
app_2-electric-1  |     params: %{
app_2-electric-1  |       "client_encoding" => "UTF8",
app_2-electric-1  |       "database" => "app_2",
app_2-electric-1  |       "user" => "electric"
app_2-electric-1  |     }
app_2-electric-1  |   }
app_2-electric-1  | ]
app_2-electric-1  | [3] 🠞 client [%PgProtocol.Message.AuthenticationMD5Password{salt: <<49, 121, 64, 128>>}]
app_2-electric-1  | 15:09:00.105 pid=<0.3407.0> proxy_session_id=3 [warning] Not validating user "electric"
app_2-electric-1  | [3] 🠜 client [
app_2-electric-1  |   %PgProtocol.Message.GSSResponse{
app_2-electric-1  |     data: <<109, 100, 53, 99, 54, 50, 101, 54, 54, 52, 48, 102, 56, 52, 54, 53,
app_2-electric-1  |       53, 97, 54, 52, 50, 55, 102, 97, 56, 97, 53, 55, 56, 53, 101, 55, 55, 49,
app_2-electric-1  |       97, 0>>
app_2-electric-1  |   }
app_2-electric-1  | ]
app_2-electric-1  | 15:09:00.108 pid=<0.3407.0> proxy_session_id=3 [debug] Starting upstream connection: []
app_2-electric-1  | [3] 🠞 client [%PgProtocol.Message.AuthenticationOk{}]
app_2-electric-1  | 15:09:00.108 pid=<0.3407.0> proxy_session_id=3 [info] Initialising injector in capture mode %Electric.Postgres.Proxy.Injector.Electric{}
app_2-electric-1  | [3] server 🠞 []
app_2-electric-1  | [3] server 🠜 []
app_2-electric-1  | 15:09:00.110 pid=<0.3407.0> proxy_session_id=3 [debug] recv_server: [Electric.Postgres.Proxy.Injector.Electric]
app_2-electric-1  | [3] 🠞 client []
app_2-electric-1  | [3] server 🠞 []
app_2-electric-1  | [3] server 🠜 []
app_2-electric-1  | 15:09:00.122 pid=<0.3407.0> proxy_session_id=3 [debug] recv_server: [Electric.Postgres.Proxy.Injector.Electric]
app_2-electric-1  | [3] 🠞 client []
app_2-electric-1  | 15:09:00.124 pid=<0.3407.0> proxy_session_id=3 [info] Upstream connection is ready to accept queries
app_2-electric-1  | [3] server 🠞 [
app_2-electric-1  |   %PgProtocol.Message.ParameterStatus{
app_2-electric-1  |     name: "application_name",
app_2-electric-1  |     value: "electric"
app_2-electric-1  |   },
app_2-electric-1  |   %PgProtocol.Message.ParameterStatus{name: "client_encoding", value: "UTF8"},
app_2-electric-1  |   %PgProtocol.Message.ParameterStatus{name: "DateStyle", value: "ISO, MDY"},
app_2-electric-1  |   %PgProtocol.Message.ParameterStatus{
app_2-electric-1  |     name: "default_transaction_read_only",
app_2-electric-1  |     value: "off"
app_2-electric-1  |   },
app_2-electric-1  |   %PgProtocol.Message.ParameterStatus{name: "in_hot_standby", value: "off"},
app_2-electric-1  |   %PgProtocol.Message.ParameterStatus{name: "integer_datetimes", value: "on"},
app_2-electric-1  |   %PgProtocol.Message.ParameterStatus{name: "IntervalStyle", value: "postgres"},
app_2-electric-1  |   %PgProtocol.Message.ParameterStatus{name: "is_superuser", value: "on"},
app_2-electric-1  |   %PgProtocol.Message.ParameterStatus{name: "server_encoding", value: "UTF8"},
app_2-electric-1  |   %PgProtocol.Message.ParameterStatus{name: "server_version", value: "14.9"},
app_2-electric-1  |   %PgProtocol.Message.ParameterStatus{
app_2-electric-1  |     name: "session_authorization",
app_2-electric-1  |     value: "postgres"
app_2-electric-1  |   },
app_2-electric-1  |   %PgProtocol.Message.ParameterStatus{
app_2-electric-1  |     name: "standard_conforming_strings",
app_2-electric-1  |     value: "on"
app_2-electric-1  |   },
app_2-electric-1  |   %PgProtocol.Message.ParameterStatus{name: "TimeZone", value: "GMT"},
app_2-electric-1  |   %PgProtocol.Message.BackendKeyData{pid: 36, key: 168058672},
app_2-electric-1  |   %PgProtocol.Message.ReadyForQuery{status: :idle}
app_2-electric-1  | ]
app_2-electric-1  | 15:09:00.124 pid=<0.3407.0> proxy_session_id=3 [debug] recv_server: [Electric.Postgres.Proxy.Injector.Electric]
app_2-electric-1  | [3] server 🠜 []
app_2-electric-1  | [3] 🠞 client [
app_2-electric-1  |   %PgProtocol.Message.ParameterStatus{
app_2-electric-1  |     name: "application_name",
app_2-electric-1  |     value: "electric"
app_2-electric-1  |   },
app_2-electric-1  |   %PgProtocol.Message.ParameterStatus{name: "client_encoding", value: "UTF8"},
app_2-electric-1  |   %PgProtocol.Message.ParameterStatus{name: "DateStyle", value: "ISO, MDY"},
app_2-electric-1  |   %PgProtocol.Message.ParameterStatus{
app_2-electric-1  |     name: "default_transaction_read_only",
app_2-electric-1  |     value: "off"
app_2-electric-1  |   },
app_2-electric-1  |   %PgProtocol.Message.ParameterStatus{name: "in_hot_standby", value: "off"},
app_2-electric-1  |   %PgProtocol.Message.ParameterStatus{name: "integer_datetimes", value: "on"},
app_2-electric-1  |   %PgProtocol.Message.ParameterStatus{name: "IntervalStyle", value: "postgres"},
app_2-electric-1  |   %PgProtocol.Message.ParameterStatus{name: "is_superuser", value: "on"},
app_2-electric-1  |   %PgProtocol.Message.ParameterStatus{name: "server_encoding", value: "UTF8"},
app_2-electric-1  |   %PgProtocol.Message.ParameterStatus{name: "server_version", value: "14.9"},
app_2-electric-1  |   %PgProtocol.Message.ParameterStatus{
app_2-electric-1  |     name: "session_authorization",
app_2-electric-1  |     value: "postgres"
app_2-electric-1  |   },
app_2-electric-1  |   %PgProtocol.Message.ParameterStatus{
app_2-electric-1  |     name: "standard_conforming_strings",
app_2-electric-1  |     value: "on"
app_2-electric-1  |   },
app_2-electric-1  |   %PgProtocol.Message.ParameterStatus{name: "TimeZone", value: "GMT"},
app_2-electric-1  |   %PgProtocol.Message.BackendKeyData{pid: 36, key: 168058672},
app_2-electric-1  |   %PgProtocol.Message.ReadyForQuery{status: :idle}
app_2-electric-1  | ]
app_2-electric-1  | [3] 🠜 client [%PgProtocol.Message.Query{query: "SELECT version();"}]
app_2-electric-1  | 15:09:00.129 pid=<0.3407.0> proxy_session_id=3 [debug] recv_client: [Electric.Postgres.Proxy.Injector.Electric]
app_2-electric-1  | [3] server 🠜 [%PgProtocol.Message.Query{query: "SELECT version()"}]
app_2-electric-1  | [3] 🠞 client []
app_2-electric-1  | [3] server 🠞 [
app_2-electric-1  |   %PgProtocol.Message.RowDescription{
app_2-electric-1  |     fields: [
app_2-electric-1  |       %PgProtocol.Message.RowDescription.Field{
app_2-electric-1  |         name: "version",
app_2-electric-1  |         oid: 0,
app_2-electric-1  |         attnum: 0,
app_2-electric-1  |         type: 25,
app_2-electric-1  |         typlen: -1,
app_2-electric-1  |         typmod: -1,
app_2-electric-1  |         fmt: 0
app_2-electric-1  |       }
app_2-electric-1  |     ]
app_2-electric-1  |   },
app_2-electric-1  |   %PgProtocol.Message.DataRow{
app_2-electric-1  |     fields: ["PostgreSQL 14.9 on x86_64-pc-linux-musl, compiled by gcc (Alpine 12.2.1_git20220924-r10) 12.2.1 20220924, 64-bit"]
app_2-electric-1  |   },
app_2-electric-1  |   %PgProtocol.Message.CommandComplete{tag: "SELECT 1"},
app_2-electric-1  |   %PgProtocol.Message.ReadyForQuery{status: :idle}
app_2-electric-1  | ]
app_2-electric-1  | [3] server 🠜 []
app_2-electric-1  | 15:09:00.130 pid=<0.3407.0> proxy_session_id=3 [debug] recv_server: [Electric.Postgres.Proxy.Injector.Operation.Simple, Electric.Postgres.Proxy.Injector.Electric]
app_2-electric-1  | [3] 🠞 client [
app_2-electric-1  |   %PgProtocol.Message.RowDescription{
app_2-electric-1  |     fields: [
app_2-electric-1  |       %PgProtocol.Message.RowDescription.Field{
app_2-electric-1  |         name: "version",
app_2-electric-1  |         oid: 0,
app_2-electric-1  |         attnum: 0,
app_2-electric-1  |         type: 25,
app_2-electric-1  |         typlen: -1,
app_2-electric-1  |         typmod: -1,
app_2-electric-1  |         fmt: 0
app_2-electric-1  |       }
app_2-electric-1  |     ]
app_2-electric-1  |   },
app_2-electric-1  |   %PgProtocol.Message.DataRow{
app_2-electric-1  |     fields: ["PostgreSQL 14.9 on x86_64-pc-linux-musl, compiled by gcc (Alpine 12.2.1_git20220924-r10) 12.2.1 20220924, 64-bit"]
app_2-electric-1  |   },
app_2-electric-1  |   %PgProtocol.Message.CommandComplete{tag: "SELECT 1"},
app_2-electric-1  |   %PgProtocol.Message.ReadyForQuery{status: :idle}
app_2-electric-1  | ]
app_2-electric-1  | [3] 🠜 client [
app_2-electric-1  |   %PgProtocol.Message.Query{
app_2-electric-1  |     query: "CREATE TABLE IF NOT EXISTS \"atdatabases_migrations_version\" (\n  id INTEGER NOT NULL PRIMARY KEY,\n  version TEXT\n);"
app_2-electric-1  |   }
app_2-electric-1  | ]
app_2-electric-1  | 15:09:00.133 pid=<0.3407.0> proxy_session_id=3 [debug] recv_client: [Electric.Postgres.Proxy.Injector.Electric]
app_2-electric-1  | [3] server 🠜 [%PgProtocol.Message.Query{query: "BEGIN"}]
app_2-electric-1  | [3] 🠞 client []
app_2-electric-1  | [3] server 🠞 [
app_2-electric-1  |   %PgProtocol.Message.CommandComplete{tag: "BEGIN"},
app_2-electric-1  |   %PgProtocol.Message.ReadyForQuery{status: :tx}
app_2-electric-1  | ]
app_2-electric-1  | 15:09:00.136 pid=<0.3407.0> proxy_session_id=3 [debug] recv_server: [Electric.Postgres.Proxy.Injector.Operation.Begin, Electric.Postgres.Proxy.Injector.Operation.Simple, Electric.Postgres.Proxy.Injector.Operation.Between, Electric.Postgres.Proxy.Injector.Electric]
app_2-electric-1  | [3] server 🠜 [
app_2-electric-1  |   %PgProtocol.Message.Query{
app_2-electric-1  |     query: "CREATE TABLE IF NOT EXISTS \"atdatabases_migrations_version\" (\n  id INTEGER NOT NULL PRIMARY KEY,\n  version TEXT\n)"
app_2-electric-1  |   }
app_2-electric-1  | ]
app_2-electric-1  | [3] 🠞 client []
app_2-electric-1  | [3] server 🠞 [
app_2-electric-1  |   %PgProtocol.Message.NoticeResponse{
app_2-electric-1  |     code: "42P07",
app_2-electric-1  |     column: nil,
app_2-electric-1  |     constraint: nil,
app_2-electric-1  |     data_type: nil,
app_2-electric-1  |     detail: nil,
app_2-electric-1  |     file: "parse_utilcmd.c",
app_2-electric-1  |     hint: nil,
app_2-electric-1  |     internal_position: nil,
app_2-electric-1  |     line: "207",
app_2-electric-1  |     message: "relation \"atdatabases_migrations_version\" already exists, skipping",
app_2-electric-1  |     position: nil,
app_2-electric-1  |     query: nil,
app_2-electric-1  |     routine: "transformCreateStmt",
app_2-electric-1  |     schema: nil,
app_2-electric-1  |     severity: "NOTICE",
app_2-electric-1  |     table: nil,
app_2-electric-1  |     where: nil
app_2-electric-1  |   },
app_2-electric-1  |   %PgProtocol.Message.CommandComplete{tag: "CREATE TABLE"},
app_2-electric-1  |   %PgProtocol.Message.ReadyForQuery{status: :tx}
app_2-electric-1  | ]
app_2-electric-1  | [3] server 🠜 [%PgProtocol.Message.Query{query: "COMMIT"}]
app_2-electric-1  | 15:09:00.138 pid=<0.3407.0> proxy_session_id=3 [debug] recv_server: [Electric.Postgres.Proxy.Injector.Operation.Simple, Electric.Postgres.Proxy.Injector.Operation.Between, Electric.Postgres.Proxy.Injector.Electric]
app_2-electric-1  | [3] 🠞 client []
app_2-electric-1  | [3] server 🠞 [
app_2-electric-1  |   %PgProtocol.Message.CommandComplete{tag: "COMMIT"},
app_2-electric-1  |   %PgProtocol.Message.ReadyForQuery{status: :idle}
app_2-electric-1  | ]
app_2-electric-1  | 15:09:00.139 pid=<0.3407.0> proxy_session_id=3 [debug] recv_server: [Electric.Postgres.Proxy.Injector.Operation.Commit, Electric.Postgres.Proxy.Injector.Operation.Pass, Electric.Postgres.Proxy.Injector.Electric]
app_2-electric-1  | [3] server 🠜 []
app_2-electric-1  | [3] 🠞 client [
app_2-electric-1  |   %PgProtocol.Message.NoticeResponse{
app_2-electric-1  |     code: "42P07",
app_2-electric-1  |     column: nil,
app_2-electric-1  |     constraint: nil,
app_2-electric-1  |     data_type: nil,
app_2-electric-1  |     detail: nil,
app_2-electric-1  |     file: "parse_utilcmd.c",
app_2-electric-1  |     hint: nil,
app_2-electric-1  |     internal_position: nil,
app_2-electric-1  |     line: "207",
app_2-electric-1  |     message: "relation \"atdatabases_migrations_version\" already exists, skipping",
app_2-electric-1  |     position: nil,
app_2-electric-1  |     query: nil,
app_2-electric-1  |     routine: "transformCreateStmt",
app_2-electric-1  |     schema: nil,
app_2-electric-1  |     severity: "NOTICE",
app_2-electric-1  |     table: nil,
app_2-electric-1  |     where: nil
app_2-electric-1  |   },
app_2-electric-1  |   %PgProtocol.Message.CommandComplete{tag: "CREATE TABLE"},
app_2-electric-1  |   %PgProtocol.Message.ReadyForQuery{status: :idle}
app_2-electric-1  | ]
app_2-electric-1  | [3] 🠜 client [%PgProtocol.Message.Query{query: "BEGIN"}]
app_2-electric-1  | 15:09:00.141 pid=<0.3407.0> proxy_session_id=3 [debug] recv_client: [Electric.Postgres.Proxy.Injector.Electric]
app_2-electric-1  | [3] server 🠜 [%PgProtocol.Message.Query{query: "BEGIN"}]
app_2-electric-1  | [3] 🠞 client []
app_2-electric-1  | [3] server 🠞 [
app_2-electric-1  |   %PgProtocol.Message.CommandComplete{tag: "BEGIN"},
app_2-electric-1  |   %PgProtocol.Message.ReadyForQuery{status: :tx}
app_2-electric-1  | ]
app_2-electric-1  | [3] server 🠜 []
app_2-electric-1  | 15:09:00.141 pid=<0.3407.0> proxy_session_id=3 [debug] recv_server: [Electric.Postgres.Proxy.Injector.Operation.Simple, Electric.Postgres.Proxy.Injector.Electric]
app_2-electric-1  | [3] 🠞 client [
app_2-electric-1  |   %PgProtocol.Message.CommandComplete{tag: "BEGIN"},
app_2-electric-1  |   %PgProtocol.Message.ReadyForQuery{status: :tx}
app_2-electric-1  | ]
app_2-electric-1  | [3] 🠜 client [
app_2-electric-1  |   %PgProtocol.Message.Query{
app_2-electric-1  |     query: "SELECT version FROM \"atdatabases_migrations_version\" WHERE id=0"
app_2-electric-1  |   }
app_2-electric-1  | ]
app_2-electric-1  | 15:09:00.143 pid=<0.3407.0> proxy_session_id=3 [debug] recv_client: [Electric.Postgres.Proxy.Injector.Electric]
app_2-electric-1  | [3] server 🠜 [
app_2-electric-1  |   %PgProtocol.Message.Query{
app_2-electric-1  |     query: "SELECT version FROM \"atdatabases_migrations_version\" WHERE id=0"
app_2-electric-1  |   }
app_2-electric-1  | ]
app_2-electric-1  | [3] 🠞 client []
app_2-electric-1  | [3] server 🠞 [
app_2-electric-1  |   %PgProtocol.Message.RowDescription{
app_2-electric-1  |     fields: [
app_2-electric-1  |       %PgProtocol.Message.RowDescription.Field{
app_2-electric-1  |         name: "version",
app_2-electric-1  |         oid: 16774,
app_2-electric-1  |         attnum: 2,
app_2-electric-1  |         type: 25,
app_2-electric-1  |         typlen: -1,
app_2-electric-1  |         typmod: -1,
app_2-electric-1  |         fmt: 0
app_2-electric-1  |       }
app_2-electric-1  |     ]
app_2-electric-1  |   },
app_2-electric-1  |   %PgProtocol.Message.CommandComplete{tag: "SELECT 0"},
app_2-electric-1  |   %PgProtocol.Message.ReadyForQuery{status: :tx}
app_2-electric-1  | ]
app_2-electric-1  | [3] server 🠜 []
app_2-electric-1  | 15:09:00.146 pid=<0.3407.0> proxy_session_id=3 [debug] recv_server: [Electric.Postgres.Proxy.Injector.Operation.Simple, Electric.Postgres.Proxy.Injector.Electric]
app_2-electric-1  | [3] 🠞 client [
app_2-electric-1  |   %PgProtocol.Message.RowDescription{
app_2-electric-1  |     fields: [
app_2-electric-1  |       %PgProtocol.Message.RowDescription.Field{
app_2-electric-1  |         name: "version",
app_2-electric-1  |         oid: 16774,
app_2-electric-1  |         attnum: 2,
app_2-electric-1  |         type: 25,
app_2-electric-1  |         typlen: -1,
app_2-electric-1  |         typmod: -1,
app_2-electric-1  |         fmt: 0
app_2-electric-1  |       }
app_2-electric-1  |     ]
app_2-electric-1  |   },
app_2-electric-1  |   %PgProtocol.Message.CommandComplete{tag: "SELECT 0"},
app_2-electric-1  |   %PgProtocol.Message.ReadyForQuery{status: :tx}
app_2-electric-1  | ]
app_2-electric-1  | [3] 🠜 client [
app_2-electric-1  |   %PgProtocol.Message.Query{
app_2-electric-1  |     query: "CREATE TABLE \"atdatabases_migrations_applied\" (\n  id BIGSERIAL NOT NULL PRIMARY KEY,\n  index INT NOT NULL,\n  name TEXT NOT NULL,\n  script TEXT NOT NULL,\n  applied_at TIMESTAMPTZ NOT NULL,\n  ignored_error TEXT NULL,\n  obsolete BOOLEAN NOT NULL\n);"
app_2-electric-1  |   }
app_2-electric-1  | ]
app_2-electric-1  | 15:09:00.148 pid=<0.3407.0> proxy_session_id=3 [debug] recv_client: [Electric.Postgres.Proxy.Injector.Electric]
app_2-electric-1  | [3] server 🠜 [
app_2-electric-1  |   %PgProtocol.Message.Query{
app_2-electric-1  |     query: "CREATE TABLE \"atdatabases_migrations_applied\" (\n  id BIGSERIAL NOT NULL PRIMARY KEY,\n  index INT NOT NULL,\n  name TEXT NOT NULL,\n  script TEXT NOT NULL,\n  applied_at TIMESTAMPTZ NOT NULL,\n  ignored_error TEXT NULL,\n  obsolete BOOLEAN NOT NULL\n)"
app_2-electric-1  |   }
app_2-electric-1  | ]
app_2-electric-1  | [3] 🠞 client []
app_2-electric-1  | [3] server 🠞 [
app_2-electric-1  |   %PgProtocol.Message.CommandComplete{tag: "CREATE TABLE"},
app_2-electric-1  |   %PgProtocol.Message.ReadyForQuery{status: :tx}
app_2-electric-1  | ]
app_2-electric-1  | [3] server 🠜 []
app_2-electric-1  | 15:09:00.174 pid=<0.3407.0> proxy_session_id=3 [debug] recv_server: [Electric.Postgres.Proxy.Injector.Operation.Simple, Electric.Postgres.Proxy.Injector.Electric]
app_2-electric-1  | [3] 🠞 client [
app_2-electric-1  |   %PgProtocol.Message.CommandComplete{tag: "CREATE TABLE"},
app_2-electric-1  |   %PgProtocol.Message.ReadyForQuery{status: :tx}
app_2-electric-1  | ]
app_2-electric-1  | [3] 🠜 client [
app_2-electric-1  |   %PgProtocol.Message.Parse{
app_2-electric-1  |     name: "",
app_2-electric-1  |     query: "UPDATE \"atdatabases_migrations_version\" SET version=$1\nRETURNING id",
app_2-electric-1  |     params: []
app_2-electric-1  |   }
app_2-electric-1  | ]
app_2-electric-1  | 15:09:00.175 pid=<0.3407.0> proxy_session_id=3 [debug] recv_client: [Electric.Postgres.Proxy.Injector.Electric]
app_2-electric-1  | [3] server 🠜 [
app_2-electric-1  |   %PgProtocol.Message.Parse{
app_2-electric-1  |     name: "",
app_2-electric-1  |     query: "UPDATE \"atdatabases_migrations_version\" SET version=$1\nRETURNING id",
app_2-electric-1  |     params: []
app_2-electric-1  |   }
app_2-electric-1  | ]
app_2-electric-1  | [3] 🠞 client []
app_2-electric-1  | [3] 🠜 client [
app_2-electric-1  |   %PgProtocol.Message.Bind{
app_2-electric-1  |     portal: "",
app_2-electric-1  |     source: "",
app_2-electric-1  |     parameters: ["5.0.2"],
app_2-electric-1  |     parameter_format_codes: [0],
app_2-electric-1  |     result_format_codes: []
app_2-electric-1  |   },
app_2-electric-1  |   %PgProtocol.Message.Describe{type: "P", name: ""},
app_2-electric-1  |   %PgProtocol.Message.Execute{portal: "", max_rows: 0}
app_2-electric-1  | ]
app_2-electric-1  | [3] server 🠜 [
app_2-electric-1  |   %PgProtocol.Message.Bind{
app_2-electric-1  |     portal: "",
app_2-electric-1  |     source: "",
app_2-electric-1  |     parameters: ["5.0.2"],
app_2-electric-1  |     parameter_format_codes: [0],
app_2-electric-1  |     result_format_codes: []
app_2-electric-1  |   },
app_2-electric-1  |   %PgProtocol.Message.Describe{type: "P", name: ""},
app_2-electric-1  |   %PgProtocol.Message.Execute{portal: "", max_rows: 0}
app_2-electric-1  | ]
app_2-electric-1  | 15:09:00.176 pid=<0.3407.0> proxy_session_id=3 [debug] recv_client: [Electric.Postgres.Proxy.Injector.Operation.Wait, Electric.Postgres.Proxy.Injector.Operation.BindExecute, Electric.Postgres.Proxy.Injector.Electric]
app_2-electric-1  | [3] 🠞 client []
app_2-electric-1  | [3] 🠜 client [%PgProtocol.Message.Sync{}]
app_2-electric-1  | [3] server 🠜 []
app_2-electric-1  | [3] 🠞 client []
app_2-electric-1  | 15:09:00.176 pid=<0.3407.0> proxy_session_id=3 [debug] recv_client: [Electric.Postgres.Proxy.Injector.Operation.Between, Electric.Postgres.Proxy.Injector.Electric]
app_2-electric-1  | 15:09:04.500 pid=<0.3404.0> origin=postgres_1 pg_client=172.23.0.2:42954 [debug] Elixir.Electric.Replication.Postgres.TcpServer: <StatusUpdate> [written_wal: #Lsn<0/1>, flushed_wal: #Lsn<0/1>, applied_wal: #Lsn<0/1>, client_timestamp: 750956944500773, immediate_response: 0]
app_2-electric-1  | 15:09:14.501 pid=<0.3404.0> origin=postgres_1 pg_client=172.23.0.2:42954 [debug] Elixir.Electric.Replication.Postgres.TcpServer: <StatusUpdate> [written_wal: #Lsn<0/1>, flushed_wal: #Lsn<0/1>, applied_wal: #Lsn<0/1>, client_timestamp: 750956954500914, immediate_response: 0]
app_2-electric-1  | 15:09:25.502 pid=<0.3404.0> origin=postgres_1 pg_client=172.23.0.2:42954 [debug] Elixir.Electric.Replication.Postgres.TcpServer: <StatusUpdate> [written_wal: #Lsn<0/1>, flushed_wal: #Lsn<0/1>, applied_wal: #Lsn<0/1>, client_timestamp: 750956965501729, immediate_response: 0]
app_2-electric-1  | 15:09:36.502 pid=<0.3404.0> origin=postgres_1 pg_client=172.23.0.2:42954 [debug] Elixir.Electric.Replication.Postgres.TcpServer: <StatusUpdate> [written_wal: #Lsn<0/1>, flushed_wal: #Lsn<0/1>, applied_wal: #Lsn<0/1>, client_timestamp: 750956976502574, immediate_response: 0]
app_2-electric-1  | 15:09:46.502 pid=<0.3404.0> origin=postgres_1 pg_client=172.23.0.2:42954 [debug] Elixir.Electric.Replication.Postgres.TcpServer: <StatusUpdate> [written_wal: #Lsn<0/1>, flushed_wal: #Lsn<0/1>, applied_wal: #Lsn<0/1>, client_timestamp: 750956986502724, immediate_response: 0]
app_2-electric-1  | 15:09:49.387 pid=<0.3087.0> [debug] Terminating idle db connection #PID<0.3091.0>
app_2-electric-1  | 15:09:49.387 pid=<0.3087.0> [debug] Terminating idle db connection #PID<0.3402.0>
app_2-electric-1  | 15:09:56.503 pid=<0.3404.0> origin=postgres_1 pg_client=172.23.0.2:42954 [debug] Elixir.Electric.Replication.Postgres.TcpServer: <StatusUpdate> [written_wal: #Lsn<0/1>, flushed_wal: #Lsn<0/1>, applied_wal: #Lsn<0/1>, client_timestamp: 750956996503350, immediate_response: 0]
app_2-postgres-1  | 2023-10-18 15:10:00.178 GMT [36] LOG:  unexpected EOF on client connection with an open transaction
```
</details>

Does anything stand out to you?

File: examples/expo/db/util.js (hunk 1 above)
Comment (magnetised) [reply examples/expo/db/util.js]: why can't we just use `localhost` here @kevin-dp ?

Comment (alco) [reply examples/expo/db/util.js]: We can, I've removed the subdomain here in a later commit. It looks like `psql`'s quirk that it even accepts such a non-standard hostname. It uses the subdomain in its prompt, as in
```
[<subdomain>] postgres:<appName>=#
```

File: examples/starter/template/db/util.js [hunk 2]
```diff
@@ -3,33 +3,48 @@ const path = require('path')
...
+const dbUser = 'electric'
+const proxyPassword = 'proxy_password'
 
-const urlComponents = DATABASE_URL.split('/')
-const DATABASE_NAME = urlComponents[urlComponents.length-1]
+// URL to use when connecting to the proxy from the host OS
+const DATABASE_URL = buildDatabaseURL(dbUser, proxyPassword, 'localhost', proxyPort, appName)
+
+// URL to use when connecting to the proxy from a Docker container. This is used when `psql` is exec'd inside the
+// `postgres` service's container to connect to the poxy running the `electric` service's container.
```
Comment (alco) [inline examples/starter/template/db/util.js:19]:
```suggestion
// `postgres` service's container to connect to the poxy running in the `electric` service's container.
```

**Comment by kevin-dp** [comment]
I can confirm the same issue as @alco is reporting. However, on my machine it only occurs every now and then (had to retry 4 times before it popped up).

**Comment by magnetised** [comment]
@kevin-dp @alco proxy fixes: https://github.com/electric-sql/electric/pull/587

**Review by icehaunter — APPROVED** [review]
LGTM

**Pull request merged**